### PR TITLE
CapturePromotion: fix a crash in case the partial_apply has type dependent operands.

### PR DESCRIPTION
**Pull request by eeckstein**
fixes rdar://problem/34219354

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci smoke test and merge

**Review by swiftix — APPROVED** [review]

**Comment by swiftix** [comment]
LGTM

File: test/SILOptimizer/capture_promotion.sil [hunk 1]
```diff
@@ -305,4 +305,24 @@ bb0(%0: $*Int, %1 : $<τ_0_0> { var τ_0_0 } <Foo>, %2 : $<τ_0_0> { var τ_0_0
   return %16 : $()
 }
 
+// CHECK-LABL: sil @test_with_dynamic_self
```
Comment (gottesmm) [inline test/SILOptimizer/capture_promotion.sil]: This has a typo:
```
CHECK-LABL
```

**Review by gottesmm — COMMENTED** [review]
Small typo.

File: test/SILOptimizer/capture_promotion.sil (hunk 1 above)
Comment (eeckstein) [reply test/SILOptimizer/capture_promotion.sil]: @gottesmm  Thanks for catching this!

Comment (gottesmm) [reply test/SILOptimizer/capture_promotion.sil]: @eeckstein No problem. Happy to help = ).

**Pull request merged**